### DISCUSSION TITLE
why can't I change the width of the left_drawer in the spa?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#2940] [Q&A] [answered]
I made some change to the SPA example ,It seems the width of the left_drawer can't be changed?

```py
#!/usr/bin/env python3
from router import Router
from nicegui import ui


@ui.page('/')  # normal index page (e.g. the entry point of the app)
@ui.page('/{_:path}')  # all other pages will be handled by the router but must be registered to also show the SPA index page
def main():
    router = Router()

    @router.add('/')
    def show_one():
        ui.label('Content One').classes('text-2xl')

    @router.add('/two')
    def show_two():
        ui.label('Content Two').classes('text-2xl')

    @router.add('/three')
    def show_three():
        ui.label('Content Three').classes('text-2xl')
        ui.space()

    # adding some navigation buttons to switch between the different pages
    **with ui.left_drawer(value=True,top_corner=True).classes(
        'column no-wrap gap-4 bg-grey-2 dark:bg-[#1b1b1b] mt-[-20px] px-8 py-12 w-fit').style('height: calc(100% + 20px) !important').props('bordered'):**
        ui.button('One', on_click=lambda: router.open(show_one)).classes('w-32')
        ui.button('Two', on_click=lambda: router.open(show_two)).classes('w-32')
        ui.button('Three', on_click=lambda: router.open(show_three)).classes('w-32')

    # this places the content which should be displayed
    router.frame().classes('w-full p-4 bg-gray-300')


ui.run()

```

**Comment by falkoschindler** [comment, marked as answer]
Hi @v479038280,

See discussion https://github.com/zauberzeug/nicegui/discussions/840 for how to set the width of a drawer:
```py
with ui.left_drawer(bordered=True).props('width=120'):
    ui.label('Hello, world!')
```

Unfortunately the "width" prop expects pixels and can't "fit" the content. But you can use some custom CSS as a workaround:
```py
ui.add_css('.q-drawer { width: fit-content !important }')

with ui.left_drawer(bordered=True):
    ui.label('Hello, world!')
```

**Reply by v479038280** [reply]
Thank you very much,  It seems that there are lots of tricky things like #2943 #1584 and this one #2940 ,which must take some extra efforts to make it work,  the nicegui documentaition could point it out,

**Reply by v479038280** [reply]
props works fine,this way has some problem  ` ui.add_css('.q-drawer { width: fit-content !important }')`  when the argument 'top_corner=True',  as shown in the picture
![微信截图_20240424175553](https://github.com/zauberzeug/nicegui/assets/25096045/35d66069-9774-410d-b542-17f4ab8aff62)

**Reply by falkoschindler** [reply]
Well, like many other things, changing the width of a drawer to a dynamic value like "fit-content" is beyond what we (or Quasar) had in mind when creating these UI elements. But with NiceGUI (and other web frameworks) there is (almost) always a way to make it work nonetheless. When answering questions like these, I could easily say "That's not possible". But I like finding workarounds, even though the solution might turn out a bit hacky.
